flink-yarn/src/test/java/org/apache/flink/yarn/AbstractYarnClusterTest.java

```diff
@@ -33,49 +33,60 @@
 import org.apache.hadoop.yarn.conf.YarnConfiguration;
 import org.apache.hadoop.yarn.exceptions.YarnException;
 import org.apache.hadoop.yarn.util.Records;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
 
 import java.io.IOException;
+import java.nio.file.Path;
 import java.util.Collections;
 import java.util.Map;
 
+import static org.apache.flink.util.TempDirUtils.newFolderIn;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
 /** Tests for the {@link YarnClusterDescriptor}. */
-public class AbstractYarnClusterTest extends TestLogger {
+@ExtendWith(TestLoggerExtension.class)
+class AbstractYarnClusterTest {
 
-    @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @TempDir private Path tempDir;
 
     /** Tests that the cluster retrieval of a finished YARN application fails. */
-    @Test(expected = ClusterRetrieveException.class)
-    public void testClusterClientRetrievalOfFinishedYarnApplication() throws Exception {
-        final ApplicationId applicationId =
-                ApplicationId.newInstance(System.currentTimeMillis(), 42);
-        final ApplicationReport applicationReport =
-                createApplicationReport(
-                        applicationId,
-                        YarnApplicationState.FINISHED,
-                        FinalApplicationStatus.SUCCEEDED);
-
-        final YarnClient yarnClient =
-                new TestingYarnClient(Collections.singletonMap(applicationId, applicationReport));
-        final YarnConfiguration yarnConfiguration = new YarnConfiguration();
-        yarnClient.init(yarnConfiguration);
-        yarnClient.start();
-
-        final YarnClusterDescriptor clusterDescriptor =
-                YarnTestUtils.createClusterDescriptorWithLogging(
-                        temporaryFolder.newFolder().getAbsolutePath(),
-                        new Configuration(),
-                        yarnConfiguration,
-                        yarnClient,
-                        false);
-
-        try {
-            clusterDescriptor.retrieve(applicationId);
-        } finally {
-            clusterDescriptor.close();
-        }
+    @Test
+    void testClusterClientRetrievalOfFinishedYarnApplication() throws Exception {
+        assertThatThrownBy(
```

Comment:
Instead of wrapping the entire test we can (and should) now target the specific line that causes the failure.

Please also check for similar instances.